### PR TITLE
optmize the logic of listNamespacesForDistributor func

### PR DESCRIPTION
**Pull request by hantmac**
Signed-off-by: hantmac <hantmac@outlook.com>

### Ⅰ. Describe what this PR does
Optmize the logic of listNamespacesForDistributor func.

### Ⅱ. Does this pull request fix one issue?

NONE

### CODE REVIEW

File: pkg/controller/resourcedistribution/utils.go [hunk 1]
```diff
@@ -293,8 +293,15 @@ func listNamespacesForDistributor(handlerClient client.Client, targets *appsv1al
...
-	} else if len(targets.NamespaceLabelSelector.MatchLabels) != 0 || len(targets.NamespaceLabelSelector.MatchExpressions) != 0 {
-		// 2. select the namespaces via targets.NamespaceLabelSelector
+	} else {
+		// 2. select the namespaces via targets.IncludedNamespaces
+		for _, namespace := range targets.IncludedNamespaces.List {
+			matchedSet.Insert(namespace.Name)
+		}
+	}
+
+	if len(targets.NamespaceLabelSelector.MatchLabels) != 0 || len(targets.NamespaceLabelSelector.MatchExpressions) != 0 {
```
Comment (veophi) [inline pkg/controller/resourcedistribution/utils.go:303]: if `target.AllNamespaces`  is selected, this `list` operation may be redundant.

Comment (hantmac) [reply pkg/controller/resourcedistribution/utils.go:303]: You mean user will set `target.AllNamespaces` and `targets.NamespaceLabelSelector` in the same time?

Comment (veophi) [reply pkg/controller/resourcedistribution/utils.go:303]: > You mean user will set `target.AllNamespaces` and `targets.NamespaceLabelSelector` in the same time?

yeah, though it is not a common settings.

**Pull request merged**